From edf0249a1705bb373fe4b26814eae16f0377be29 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Sun, 17 Apr 2016 12:33:51 +0530
Subject: [PATCH] Add support for loading GIO modules from the distro's GIO
 module path

This is needed to e.g. load GVfs from the distro. Without this elements
like giosrc are pretty useless and we really don't want to ship GVfs.

Original patch by sebastian.droege@collabora.co.uk; blindly ported to 2.48 by
Nirbheek
---
 gio/giomodule.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index da7c167..b97aa78 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1084,6 +1084,11 @@ _g_io_modules_ensure_loaded (void)
       g_io_modules_scan_all_in_directory_with_scope (module_dir, scope);
       g_free (module_dir);
 
+#ifdef GST_SDK_GLIB_GIO_DISTRO_GIO_MODULE_PATH
+      /* Now load all modules from the distro, e.g. gvfs */
+      g_io_modules_scan_all_in_directory_with_scope (GST_SDK_GLIB_GIO_DISTRO_GIO_MODULE_PATH, scope);
+#endif
+
       g_io_module_scope_free (scope);
 
       /* Initialize types from built-in "modules" */
-- 
2.7.2.windows.1

